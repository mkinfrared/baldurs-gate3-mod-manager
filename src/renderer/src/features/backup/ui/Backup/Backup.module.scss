.Backup {
  align-items: center;
  display: flex;
  gap: var(--spacing-3);
  padding: var(--spacing-5);

  .restore {
    margin-left: auto;
  }

  &:not(:last-child) {
    border-bottom: none;
  }
}
