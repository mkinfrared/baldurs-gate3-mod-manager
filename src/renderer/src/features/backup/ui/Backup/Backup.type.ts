import { Game } from "@renderer/shared/types";

export type BackupProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  name: string;
  filePath: string;
  game: Game;
};
