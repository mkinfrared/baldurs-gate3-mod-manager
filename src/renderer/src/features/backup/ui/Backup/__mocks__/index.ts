export { BackupMock as Backup } from "./Backup.mock";
