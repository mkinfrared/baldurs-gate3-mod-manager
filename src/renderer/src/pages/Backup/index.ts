export { Backup } from "./Backup";

export * from "./Backup.type";
