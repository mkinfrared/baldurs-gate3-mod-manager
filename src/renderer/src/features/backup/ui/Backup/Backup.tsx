import { delay } from "@common/lib";
import Copy from "@renderer/shared/assets/icons/copy.svg";
import Delete from "@renderer/shared/assets/icons/delete-forever.svg";
import Restore from "@renderer/shared/assets/icons/restore.svg";
import Tick from "@renderer/shared/assets/icons/tick.svg";
import { classNames, trpc } from "@renderer/shared/lib/helpers";
import { Card, IconButton, Text } from "@renderer/shared/ui";

import css from "./Backup.module.scss";
import { BackupProps } from "./Backup.type";

const Backup = ({ className, filePath, game, name }: BackupProps) => {
  const utils = trpc.useUtils();

  const { mutate: restore, isLoading: isRestoring } =
    trpc.backup.restoreFromFile.useMutation({
      onMutate: async (variables) => {
        await delay(850);

        return variables;
      },
      onSettled: () => {
        utils.mod.invalidate();
      },
    });

  const { mutate: deleteBackup, isLoading: isDeleting } =
    trpc.backup.deleteBackup.useMutation({
      onMutate: async (variables) => {
        await delay(850);

        return variables;
      },
      onSettled: () => {
        utils.backup.getBackups.invalidate();
      },
    });

  const {
    mutateAsync: copy,
    isLoading: isCopying,
    isSuccess: isCopied,
  } = trpc.backup.copyBackupContent.useMutation({
    onMutate: async (variables) => {
      await delay(850);

      return variables;
    },
    onSuccess: async (content) => {
      if (content) {
        await navigator.clipboard.writeText(content);
      }
    },
  });

  return (
    <Card className={classNames(css.Backup, className)} data-testid="Backup">
      <Text>{name}</Text>
      <IconButton
        className={css.copy}
        onClick={() => copy(filePath)}
        loading={isCopying}
        disabled={isCopying}
        title="Copy content to buffer"
      >
        {isCopied ? <Tick /> : <Copy />}
      </IconButton>
      <IconButton
        onClick={() => restore({ file: filePath, gameKey: game })}
        loading={isRestoring}
        disabled={isRestoring}
        title="Restore modsettings from this backup"
      >
        <Restore />
      </IconButton>
      <IconButton
        color="error"
        onClick={() => deleteBackup(filePath)}
        loading={isDeleting}
        disabled={isDeleting}
        title="Delete from file system"
      >
        <Delete />
      </IconButton>
    </Card>
  );
};

export { Backup };
