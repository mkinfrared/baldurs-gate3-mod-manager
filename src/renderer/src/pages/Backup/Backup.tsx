import { classNames } from "@renderer/shared/lib/helpers";
import { Heading } from "@renderer/shared/ui";
import { BackupSection } from "@renderer/widgets";

import css from "./Backup.module.scss";
import { BackupProps } from "./Backup.type";

const Backup = ({ className, game }: BackupProps) => (
  <div className={classNames(css.Backup, className)} data-testid="Backup">
    <Heading variant="h1">Backups</Heading>
    <BackupSection game={game} />
  </div>
);

export { Backup };
