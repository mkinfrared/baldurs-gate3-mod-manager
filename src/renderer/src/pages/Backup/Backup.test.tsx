import { render } from "@testing-library/react";

import { Backup } from "./Backup";

vi.mock("@renderer/widgets");

describe("<Backup />", () => {
  const game = "BG3";
  const Component = <Backup game={game} />;

  it("should be defined", () => {
    expect(Backup).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Backup");

    expect(element).toBeDefined();
  });
});
