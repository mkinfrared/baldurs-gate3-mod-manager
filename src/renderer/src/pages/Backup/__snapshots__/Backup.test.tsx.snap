// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Backup /> > should match the snapshot 1`] = `
<div>
  <div
    class="Backup"
    data-testid="Backup"
  >
    <h1
      class="Heading"
      data-testid="Heading"
    >
      Backups
    </h1>
  </div>
</div>
`;
